### DISCUSSION TITLE
How can I set proxy_http_version dynamically per upstream request?

### DISCUSSION THREAD

**Discussion by mssaisandeep** [Kong/kong#7330] [Ideas and feature requests]
**Requirement:**  Before upstreaming the request, I would like to change the http request protocol version. I couldn't find any API's (kong/nginx) to set the proxy_http_version. I also tried adding content_by_lua_block as mentioned in this link: https://stackoverflow.com/questions/64301671/how-to-set-proxy-http-version-in-lua-code-before-upstreaming-the-request-in-ngin. But content_by_lua_block is not even getting executed. 

Please help me if there is any way that we can acheive this. Let me know if you need more information. 

@thibaultcha @bungle

**Comment by bungle** [comment]
@mssaisandeep, so you basically want to proxy with `http 1.0`?

**Comment by mssaisandeep** [comment]
@bungle Yes, I want to set it to http1.0 to the upstream request **dynamically**. 
I have 2 services which are running behind kong. One can work with http1.1 and the other should require the request to be http1.0. 

Do you need more information?

**Comment by bungle** [comment]
@mssaisandeep no, I have all the information. I can propose a feature, but I don't want to guarantee that it gets approved/merged. We'll see how it goes.

**Comment by mssaisandeep** [comment]
Thank you @bungle. 

FYI, I have been using kong as a gateway/proxy for the last 2 years for all kinds of applications (HTTPs/RDP/SSH). I solved various use cases to support legacy as well as other different kinds of HTTP(s) applications behind kong. If you need it, I can pick the best one out of that list (which I added support using plugins). That may be helpful and maybe you can add direct support in the kong itself.

**Reply by Sam123ben** [reply]
Hi team, any update on it as we too need a similar soln too where two services require diff way where one with the default 1.1 and other 1.0.

Can someone please help me with how to configure it in kong so that the ingress can dynamically work and resolve the chunked.

**7400 upstream sent invalid chunked response while reading upstream**

**Comment by oldthreefeng** [comment]
any solution for this?  @mssaisandeep  @Sam123ben
